core/src/main/java/com/linecorp/armeria/server/ReflectiveDependencyInjector.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server;
+
+import static org.reflections.ReflectionUtils.getConstructors;
+import static org.reflections.ReflectionUtils.withParametersCount;
+
+import java.lang.reflect.Constructor;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.Iterables;
+
+import com.linecorp.armeria.common.annotation.Nullable;
+
+final class ReflectiveDependencyInjector implements DependencyInjector {
+
+    private static final Logger logger = LoggerFactory.getLogger(ReflectiveDependencyInjector.class);
+
+    private static final Map<Class<?>, Object> instances = new HashMap<>();
```

Comment:
If this class is not a singleton, should we keep `instances` as a member field?

Reply:
Thanks, changed to the member field. 😉